DependencyInjection/Compiler/CacheCompatibilityPass.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+namespace Doctrine\Bundle\DoctrineBundle\DependencyInjection\Compiler;
+
+use Doctrine\Common\Cache\CacheProvider;
+use Doctrine\Common\Cache\Psr6\CacheAdapter;
+use Doctrine\Common\Cache\Psr6\DoctrineProvider;
+use Psr\Cache\CacheItemPoolInterface;
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\DependencyInjection\Reference;
+
+use function array_keys;
+use function is_a;
+use function trigger_deprecation;
+
+/** @internal  */
+final class CacheCompatibilityPass implements CompilerPassInterface
+{
+    private const CACHE_METHODS_PSR6_SUPPORT_MAP = [
+        'setMetadataCache' => true,
+        'setQueryCacheImpl' => false,
+        'setResultCacheImpl' => false,
+    ];
+
+    public function process(ContainerBuilder $container): void
+    {
+        foreach (array_keys($container->findTaggedServiceIds(IdGeneratorPass::CONFIGURATION_TAG)) as $id) {
```

Comment:
I dislike the coupling to another compiler pass. If this is a generic setting, we should probably move this to the extension class or an interface for passes that defines these constants and use that, alternatively I'm also fine with this class having its own constant.

Reply:
I think we don't have an interface for these constants. Feel free to create something like that (perhaps just putting it to DoctrineExtension would be enough), although I feel it's a bit outside the scope of this PR. But keep in mind it should be BC change (can't remove old constant) and it's a constant for internal use only.

Reply:
Inlined the constant for now and will revisit in a separate PR.